sdks/python/apache_beam/transforms/trigger.py

```diff
@@ -378,6 +378,8 @@ class AfterCount(TriggerFn):
   COUNT_TAG = _CombiningValueStateTag('count', combiners.CountCombineFn())
 
   def __init__(self, count):
+    if count == 0:
+      raise ValueError("count (%s) must be a non-zero natural number." % count)
```

Comment:
<!--new_thread; commit:17604fb4ecfba607daa3c22312c02865b02ecee7; resolved:0-->
- Make the error statement match the condition. The condition is not checking for natural numbers either. For example `2.3` would not raise this error even though it is not a natural number. You can rephrase it as `must be a positive number`.

- For string formatting of numbers, you can use `%d` instead of `%s`.